### DISCUSSION TITLE
Schema fields that start with underscore should be hidden in the API response

### DISCUSSION THREAD

**Discussion by subnetmarco** [Kong/kong#7199] [Ideas and feature requests]
In some situations it's very convenient to cache some otherwise long operations inside the datastore in specific private fields, like https://github.com/Mashape/kong/blob/master/kong/plugins/ssl/schema.lua#L27-L29

This is to avoid the "Dogpile Effect" in production, and increase performance.

This fields are currently being displayed in the API response, while they should only be visibile internally. Not sure if we want to use the underscore convention, or have a specific `hide = true` property for the field.

**Comment by thibaultcha** [comment]
Sounds good. I'm more in favour of the `hide = true` than the underscore, just to stay consistent with how fields acquire properties.

**Comment by subnetmarco** [comment]
@thibaultCha this needs to be implemented in the Admin API right before returning the response, because we still want to have the visibility of the fields when the DAO is being invoked.

**Comment by subnetmarco** [comment]
Actually `hide = true` is the wrong property name. Should be `internal = true`.
